nc-games/src/Components/NewComment.jsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from "react";
+import { postComment } from "../api";
+
+function NewComment({reviewId, hasPostedComment, setHasPostedComment}) {
+    const [newComment, setNewComment] = useState("");
+    const [isPostingError, setIsPostingError] = useState(false)
+    const [hasPosted, setHasPosted] = useState(false)
+    const [hasPostedError, setHasPostedError] = useState(false)
+
+    function handleSubmit(event) {
+        event.preventDefault()
+        setHasPostedComment(false);
+        setHasPostedError(false);
+        if(!hasPosted) {
+            setHasPosted(true);
+            postComment(reviewId, newComment).then(response => {
+                setHasPosted(false)
+                setHasPostedComment(true)
```

Comment:
Please implement optimistic rendering of comments. At the moment they only display when posting is done.